pkg/controller/manage/manage_controller.go

```diff
@@ -203,6 +126,164 @@ func NewKmeshManageController(client kubernetes.Interface, security *kmeshsecuri
 	}, nil
 }
 
+func handlePodAddFunc(obj interface{}, namespaceLister v1.NamespaceLister, queue workqueue.RateLimitingInterface, security *kmeshsecurity.SecretManager) {
+	pod, ok := obj.(*corev1.Pod)
+	if !ok {
+		log.Errorf("expected *corev1.Pod but got %T", obj)
+		return
+	}
+
+	namespace, err := namespaceLister.Get(pod.Namespace)
+	if err != nil {
+		log.Errorf("failed to get pod namespace %s: %v", pod.Namespace, err)
+		return
+	}
+
+	if !utils.ShouldEnroll(pod, namespace) {
+		if pod.Annotations[constants.KmeshRedirectionAnnotation] == "enabled" {
+			disableKmeshManage(pod, queue, security)
+		}
+		return
+	}
+	enableKmeshManage(pod, queue, security)
+}
+
+func handlePodUpdateFunc(oldObj, newObj interface{}, namespaceLister v1.NamespaceLister, queue workqueue.RateLimitingInterface, security *kmeshsecurity.SecretManager) {
+	oldPod, okOld := oldObj.(*corev1.Pod)
+	newPod, okNew := newObj.(*corev1.Pod)
+	if !okOld || !okNew {
+		log.Errorf("expected *corev1.Pod but got %T and %T", oldObj, newObj)
+		return
+	}
+
+	namespace, err := namespaceLister.Get(newPod.Namespace)
+	if err != nil {
+		log.Errorf("failed to get pod namespace %s: %v", newPod.Namespace, err)
+		return
+	}
+
+	// enable kmesh manage
+	if oldPod.Annotations[constants.KmeshRedirectionAnnotation] != "enabled" && newPod.Annotations[constants.KmeshRedirectionAnnotation] != "enabled" && utils.ShouldEnroll(newPod, namespace) {
```

Comment:
I feel we can check 

`if newPod.Annotations[constants.KmeshRedirectionAnnotation] != "enabled" && utils.ShouldEnroll(newPod, namespace)`, not sure why we didnot